jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/internal/ContentCopier.java

```diff
@@ -57,7 +58,7 @@ protected Action process() throws Throwable
 
         if (current == null)
         {
-            source.demand(this::succeeded);
+            source.demand(Invocable.from(getInvocationType(), this::succeeded));
```

Comment:
No, the nested callback might be blocking.

Reply:
Should the invocationType be that of the callback?